### PR TITLE
ncrack: update license with exceptions

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/ncrack.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class Ncrack < Formula
   desc "Network authentication cracking tool"
   homepage "https://nmap.org/ncrack/"
-  license "GPL-2.0-only"
+  # The program license includes other non-representable exceptions in addition
+  # to the OpenSSL exception. These can be seen in the installed COPYING file.
+  license "GPL-2.0-only" => { with: "openvpn-openssl-exception" }
```
Comment (cho-m) [inline Formula/ncrack.rb:6]: Not sure if better to go with this or `:cannot_represent`.

All exceptions must be accepted https://github.com/nmap/ncrack/blob/master/COPYING#L9-L12
```
 * This program is free software; you may redistribute and/or modify it    *
 * under the terms of the GNU General Public License as published by the   *
 * Free Software Foundation; Version 2 ("GPL"), BUT ONLY WITH ALL OF THE   *
 * CLARIFICATIONS AND EXCEPTIONS DESCRIBED HEREIN.  This guarantees your   *
```

OpenSSL exception https://github.com/nmap/ncrack/blob/master/COPYING#L60-L64
```
 * As another special exception to the GPL terms, the Nmap Project grants  *
 * permission to link the code of this program with any version of the     *
 * OpenSSL library which is distributed under a license identical to that  *
 * listed in the included docs/licenses/OpenSSL.txt file, and distribute   *
 * linked combinations including the two.                                  *
```

---

Fedora just refers to this as **with exceptions** (https://src.fedoraproject.org/rpms/ncrack/blob/rawhide/f/ncrack.spec#_8)

Debian just refers to this as **with nmap exception** (https://metadata.ftp-master.debian.org/changelogs//main/n/ncrack/ncrack_0.7+debian-3_copyright)

Neither is a part of SPDX.

Comment (carlocab) [reply Formula/ncrack.rb:6]: @Homebrew/plc?

Comment (SMillerDev) [reply Formula/ncrack.rb:6]: I'd say `:cannot_represent`, clearly without our SPDX requirement it's already hard to represent.

Comment (MikeMcQuaid) [reply Formula/ncrack.rb:6]: Agreed on `:cannot_represent`.

**Review by fxcoudert — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**